src/secret.rs

```diff
@@ -0,0 +1,112 @@
+//! This module implements secrets in the form of protected memory.
+
+use secp256k1::{Error as Secp256k1Error, SecretKey};
+use std::fmt::{self, Debug, Formatter};
+use std::ops::Deref;
+use std::str::FromStr;
+use web3::types::Address;
+use zeroize::Zeroizing;
+
+/// A secret key used for signing and hashing.
+///
+/// This type has a safe `Debug` implementation that does not leak information.
+/// Additionally, it implements `Drop` to zeroize the memory to make leaking
+/// passwords less likely.
+#[derive(Clone)]
+pub struct PrivateKey(SecretKey);
+
+impl PrivateKey {
+    /// Creates a new private key from raw bytes.
+    pub fn from_raw(raw: [u8; 32]) -> Result<Self, Secp256k1Error> {
+        PrivateKey::from_slice(&raw)
+    }
+
+    /// Creates a new private key from a slice of bytes.
+    pub fn from_slice<B: AsRef<[u8]>>(raw: B) -> Result<Self, Secp256k1Error> {
+        Ok(PrivateKey(SecretKey::from_slice(raw.as_ref())?))
+    }
+
+    /// Creates a new private key from a hex string representation. Accepts hex
+    /// string with or without leading `"0x"`.
+    pub fn from_hex_str<S: AsRef<str>>(s: S) -> Result<Self, Secp256k1Error> {
+        let hex_str = {
+            let s = s.as_ref();
+            if s.starts_with("0x") {
+                &s[2..]
+            } else {
+                s
+            }
+        };
+        let key = SecretKey::from_str(hex_str)?;
+
+        Ok(PrivateKey(key))
+    }
+
+    /// Gets the public address for a given private key.
+    pub fn public_address(&self) -> Address {
+        todo!()
```

Comment:
This is needed for dex-service integration I believe

Reply:
It is not, I meant to create a draft PR. There are also some unit tests that are needed.